tests/test_smoke.py

```diff
@@ -3214,68 +3235,106 @@ def test_skyserve_cancel():
 
 
 @pytest.mark.gcp
-@pytest.mark.sky_serve
+@pytest.mark.serve
 def test_skyserve_update():
     """Test skyserve with update"""
     name = _get_service_name()
     test = Test(
         f'test-skyserve-update',
         [
             f'sky serve up -n {name} -y tests/skyserve/update/old.yaml',
-            _SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=1),
-            f'{_get_serve_endpoint(name)}; curl -L http://$endpoint | grep "Hi, SkyPilot here"',
+            _SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=2),
+            f'{_SERVE_ENDPOINT_WAIT.format(name=name)}; curl -L http://$endpoint | grep "Hi, SkyPilot here"',
+            f'sky serve update {name} --mode blue_green -y tests/skyserve/update/new.yaml',
+            # sleep before update is registered.
+            'sleep 20',
+            f'{_SERVE_ENDPOINT_WAIT.format(name=name)}; '
+            'until curl -L http://$endpoint | grep "Hi, new SkyPilot here!"; do sleep 2; done;'
+            # Make sure the traffic is not mixed
+            'curl -L http://$endpoint | grep "Hi, new SkyPilot here"',
+            # The latest 2 version should be READY and the older versions should be shutting down
+            _check_replica_in_status(name, [(2, False, 'READY'),
+                                            (2, False, 'SHUTTING_DOWN')]),
+        ],
+        _TEARDOWN_SERVICE.format(name=name),
+        timeout=20 * 60,
+    )
+    run_one_test(test)
+
+
+@pytest.mark.aws
+@pytest.mark.gcp
+@pytest.mark.serve
+def test_skyserve_rolling_update():
+    """Test skyserve with update"""
+    name = _get_service_name()
+    test = Test(
+        f'test-skyserve-update',
+        [
+            f'sky serve up -n {name} -y tests/skyserve/update/old.yaml',
+            _SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=2),
+            f'{_SERVE_ENDPOINT_WAIT.format(name=name)}; curl -L http://$endpoint | grep "Hi, SkyPilot here"',
             f'sky serve update {name} -y tests/skyserve/update/new.yaml',
             # sleep before update is registered.
             'sleep 20',
-            _SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=1),
-            f'{_get_serve_endpoint(name)}; curl -L http://$endpoint | grep "Hi, new SkyPilot here!"',
+            _SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=2),
```

Comment:
In this example, it will be 2/4 first. 
2 ready old replicas. 2 provisioning new replicas.
-> 
2 shutting down old replicas. 2 ready new replicas
->
2 ready new replicas.

By the time we reach `_SERVE_WAIT_UNTIL_READY.format(name=name, replica_num=2),`, all the replicas will be ready? 
This behavior is similar to blue-green update?

Reply:
Added the replica id env var and sleep for the replicas with even id, so that the period of the mixing traffic will be longer.